packages/pipeline/src/scripts/pull_exchange_transactions_from_etherscan.ts

```diff
@@ -0,0 +1,70 @@
+import { Connection, ConnectionOptions, createConnection, Repository } from 'typeorm';
+
+import { getContractAddressesForNetworkOrThrow, NetworkId } from '@0x/contract-addresses';
+import { web3Factory } from '@0x/dev-utils';
+import { Web3ProviderEngine } from '@0x/subproviders';
+import { logUtils } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+
+import { EtherscanSource } from '../data_sources/etherscan';
+import { EtherscanTransaction } from '../entities';
+import * as ormConfig from '../ormconfig';
+import { parseEtherscanTransactions } from '../parsers/etherscan';
+import { EXCHANGE_START_BLOCK, handleError, INFURA_ROOT_URL } from '../utils';
+
+const BATCH_SAVE_SIZE = 1000; // Number of orders to save at once.
+const START_BLOCK_OFFSET = 100; // Number of blocks before the last known block to consider when updating fill events.
+const BLOCK_FINALITY_THRESHOLD = 10; // When to consider blocks as final. Used to compute default endBlock.
+
+// API key to use if environment variable has not been set
+const FALLBACK_API_KEY = 'YourApiKeyToken';
+
+let connection: Connection;
+
+(async () => {
+    let apiKey = process.env.ETHERSCAN_API_KEY;
+    if (apiKey === undefined) {
+        logUtils.log(`Missing env var: ETHERSCAN_API_KEY - using default API key: ${FALLBACK_API_KEY}`);
```

Comment:
I think if api key is undefined we should just throw an exception, so that we see the failure on airflow.